### PR TITLE
Form Validation

### PR DESCRIPTION
**Pull request by zbarnz**
### Context
Checks if username and email exist when user is registering in real time.
### Implementation
Added a function that makes a request to the database on every keystroke. The form messages will need styled but wanted to submit this for review before because I'm procrastinating.

### Merge Checklist
- [x] Correct Target Branch
- [x] Pre-Review Diff Check
- [x] PR Description
- [x] Add Reviewers and Assignees
- [ ] Review Complete
- [ ] Rebase
- [ ] Rebase Approved

═══════════════════░▒░══════════════════
════════════▒▒▒▓███████████▓▒═══════════
════════▒██████████▓▓▓▓██▓████▓═════════
══════▒███████▓█▓▓▓█▓█▓███▓▓▓███════════
═════▓███▓██▓▓████▓██▓█▓▓▓███▓██▒═══════
═════▓██▓███▓█▓████████▓▓█████▓██═══════
═════▒█████████▓▓▓███████████████▒══════
═════▒██████▓█████▒▒▒▒▒██████████▒══════
═════▒███████▓█▓█▒═════░▒▓█▓▓▒▒██▒══════
═════▒█████████▓█▓░░▒▒░░░░░▒▒▒▒██▒══════
═════▒████████████▒░▒▒▒▒▒▒▒▒▒▒▒▓█▒══════
═════▒██████████▓▓▓▒▒▒▒▒▒▒▒▒▒▒░▓█▒══════
═════▒███████▓███▓▒▒▒▒▒▒▒▒▒▒▒▒░▒▓▒══════
═════▓█▒░░▓██▓█▓▒▒▒▒░░░░░░░░░░░▒█▒══════
═════▓█▒▒▒░▒███▒░░▒░░░▒▒▒▒▒█▓█████══════
═════██░▒██▒▒████▒░▓██████████████══════
═════▒█▒▒▒▓▒░▓█▓███████████▒▒█████══════
══════▒█═░░▒▒▒█▒▒▒▒█▓██████░═████═══════
═▒▓███▓██░░▒▒▒▒▒▒▒▒██████▓▒░░▒██▒═══════
██████████▓█▒▒▒▒▒▒▒▒█▓▒▒▒▒░░░═░█════════
█████████▓▓█▓▒▒▒▒▒▓▒▒▒▒▒▒▒▓▓▒░▒█════════
██████████▒▒▓▓▒▒▒▒▒▒▒▒▒▒░▒▓██▓▒█════════
▓▓█▓████▓▒▒▒▒▓▓▒▒▒▒▒▒▒▒▒▓▒▒▒▓▓▓██═══════
██▓▒██▒░░▒▓▒▒▒▓█▓▒▒▒▒▒▒█████▒▒▓███══════
█▓▒█▓░░▒▒▒▒▒▒▒▒▒▓▓▒▒▓▒▒░░░▒▒▒▒█████═════
▓▓██░░▒▒▒▒▒▒▒▒▓▒▒▓▓▓▒▒▒▒▒▓██▓▓██████════
▓██▒░▒▒▒▒▒▒▒▒▒▓▓▓▓▓██▓▒▒▒░▒░▒█████▓██═══
▓▓█▒░▒▒▒▒▒▒▓▒▒▒▓▓██████▓▒▒▒▒█████▓▓██░══
█▓█▒░▒▒▒▒▒▒▒▒▒▓▓▓▓████████▓▒████▓██▓██══
▓██▒░▒░▒░░░▒▒▒▒▒▒▒▓▓████▓▒▒▒███▓▓█▓▓▓█══
▒██░░░░░░░░▒▒▒▒▒▒▒▓▓███▓▓▒▒██▓█▓█▓▒█▓██░
▒█▒═░░░░░▒▒░▒▒▒▒▒▒▒▒▒▒▒▓▒▒██▓▓▒█▒▒█▓▒▒▓█
▒█▓░░░░░░░▒▒░▒▒▒▒░░░░▒▒░░██▒▓▒▓▒▒█▓▒▒▒▒█
▒▒█▒═░░░░░░░░░░░░░░░▒░═▒█▒▒▓▒▓█▓▓▒▒▒▒▒▒▓
█▒█▒═░░░░░░░░░░░░░░▒═░██▒▒▒▓▓▒▒▓▒▒▒▒▒▓▒▓
▓▓▒█░░░░░░░░░░░░░░░═▒█▒▒██▓▒▒▓▒▒▒▒▒▓▓▓▒▓
▓█▒█▒═░░░░░░░░▒░░░═▒█▒▒▓▓▓▓▓▒▒▓▒▒▓▓▓▓▒▓▓
▓█▓▒▒═░░░░░▒░▒░░▒═▒█▒▓▓▓▓▓█▓▒▒█▓▒▓▓▓▒▓▓▒
▓▓█▓▒░░░░░░░▒▒▒▒═░█▓▓▓▓▓▓▓▓▓▓▒▓██▒▒▒▓▓█▓
▓▓▓▓█▒═░░░░▒▒▒▒═░▓▓▓▓▓▒▓███▒▓█▒▒▓█▓▒▒▒▒▒
▓▓▒▓█▓═░░▒▒░▒▒═░██▓▓▓▓▓████▒▓▓██▓▓██▓▓▓▓
▓▒▒▓▒█░░▒▒░▒░═▒█████████▓███▓▓▓▓▓▓▓▓████
▓█▓▓▓█▓═▒░▒░░██████████████▓████████████
▓█▓▓▓▒█░░▒═░██████▓█▓█▓█▓▓▓▒█▒▒▒▒▒██████

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -103,7 +103,7 @@
     ],
     "linebreak-style": [
       2,
-      "unix"
```
Comment (thudsonbu) [inline .eslintrc:106]: Illegal

File: www/pages/register.js [hunk 2]
```diff
@@ -18,13 +21,56 @@ class Register extends Component {
...
+        this.setState({
+          showEmailError: false
+        });
+      }
+
+      console.log( result );
+    });
   }
 
   handleSubmit( event ) {
```
Comment (thudsonbu) [inline www/pages/register.js:76]: We should disable the submit button until the two checks have passed.

File: www/pages/register.js [hunk 3]
```diff
@@ -18,13 +21,56 @@ class Register extends Component {
...
     });
+
+  }
+
+  handleChangeUsername( event ) {
+    this.setState({
+      [ event.target.name ]: event.target.value
+    });
+
+    checkExistingUsername( event.target.value )
```
Comment (thudsonbu) [inline www/pages/register.js:41]: What do you think about calling it `userNameUniqueness`.

File: www/pages/register.js (hunk 2 above)
Comment (thudsonbu) [reply www/pages/register.js:76]: To do that you would need to add three values to the state `loading`, `uniqueUser`, and `uniqueEmail`. While the checks are running you set `loading` to true. The submit button is disabled while `loading` is true or while either `uniqueUser` or `uniqueEmail is false.`

File: www/pages/register.js (hunk 3 above)
Comment (zbarnz) [reply www/pages/register.js:41]: I dont like it

**Review by Dahvd — APPROVED** [review]
Cool

File: www/components/Button.js [hunk 4]
```diff
@@ -14,7 +14,12 @@ const Button = ({
   type = 'neutral',
   stretch,
   children,
+<<<<<<< HEAD
```
Comment (thudsonbu) [inline www/components/Button.js:17]: 😡

File: www/components/Button.js [hunk 5]
```diff
@@ -41,7 +46,12 @@ const Button = ({
       buttonStyles += ' ' + styles.outlined;
       break;
     case 'disabled':
+<<<<<<< HEAD
       buttonStyles += ' ' + styles.disabled;
+=======
```
Comment (thudsonbu) [inline www/components/Button.js:51]: 😡

**Review by thudsonbu — CHANGES_REQUESTED** [review]
You might want to remove `www/components/Button.js`

**Review by thudsonbu — APPROVED** [review]

**Pull request merged**